src/mpi/romio/adio/ad_lustre/ad_lustre_wrcoll.c

```diff
@@ -383,15 +375,16 @@ static void ADIOI_LUSTRE_Exch_and_write(ADIO_File fd, const void *buf,
      * at least another 8Mbytes of temp space is unacceptable.
      */
 
-    int hole, i, j, m, flag, ntimes = 1, max_ntimes, buftype_is_contig;
+    int hole, i, m, flag, ntimes = 1, buftype_is_contig;
+    MPI_Count max_ntimes, req_len, send_len;
     ADIO_Offset st_loc = -1, end_loc = -1, min_st_loc, max_end_loc;
     ADIO_Offset off, req_off, send_off, iter_st_off, *off_list;
     ADIO_Offset max_size, step_size = 0;
-    int real_size, req_len, send_len;
-    int *recv_curr_offlen_ptr, *recv_count, *recv_size;
-    int *send_curr_offlen_ptr, *send_size;
-    int *sent_to_proc, *recv_start_pos;
-    int *curr_to_proc, *done_to_proc;
+    int real_size;
+    MPI_Count *recv_count, *send_curr_offlen_ptr, *recv_curr_offlen_ptr;
+    MPI_Count *recv_size, *send_size;
```

Comment:
Just a comment.

For Lustre, each round of two-phase I/O only writes to a file striping unit size (of type `__u32`)
per I/O aggregator. Thus `send_size` and `recv_size` will never exceed UINT_MAX.
Based on that, some of the counters are fine with being `int`, such as `srt_len` and `srt_num`.

I guess changing `int` to `MPI_Count` can silence the compile warnings on 64-to-32 int.

Reply:
that's exactly right.  There were so many shorten-64-to-32 warnings that I could not tell which ones were fine (like this one) and which were going to overflow.